silkrpc/core/blocks.cpp

```diff
@@ -47,10 +52,44 @@ boost::asio::awaitable<uint64_t> get_highest_block_number(const core::rawdb::Dat
 }
 
 boost::asio::awaitable<uint64_t> get_latest_block_number(const core::rawdb::DatabaseReader& reader) {
+    const auto kv_pair = co_await reader.get(silkrpc::db::table::kLastForkchoice, silkworm::bytes_of_string(std::string("headBlockHash")));
+    const auto head_block_hash_data = kv_pair.value;
+    if (!head_block_hash_data.empty()) {
+        const auto head_block_hash = silkworm::to_bytes32(head_block_hash_data);
+        const silkworm::BlockHeader head_block_header = co_await rawdb::read_header_by_hash(reader, head_block_hash);
+        co_return head_block_header.number;
+    }
+
     const auto latest_block_number = co_await stages::get_sync_stage_progress(reader, stages::kExecution);
     co_return latest_block_number;
 }
 
+boost::asio::awaitable<uint64_t> get_forkchoice_finalized_block_number(const core::rawdb::DatabaseReader& reader) {
+    const auto kv_pair = co_await reader.get(silkrpc::db::table::kLastForkchoice, silkworm::bytes_of_string(std::string("finalizedBlockHash")));
+    const auto finalized_block_hash_data = kv_pair.value;
+    if (finalized_block_hash_data.empty()) {
+        SILKRPC_LOG << "no finalized forkchoice block number found\n";
+        co_return 0;
+    }
+    const auto finalized_block_hash = silkworm::to_bytes32(finalized_block_hash_data);
+
+    const auto finalized_header = co_await rawdb::read_header_by_hash(reader, finalized_block_hash);
+    co_return finalized_header.number;
+}
+
+boost::asio::awaitable<uint64_t> get_forkchoice_safe_block_number(const core::rawdb::DatabaseReader& reader) {
+    const auto kv_pair = co_await reader.get(silkrpc::db::table::kLastForkchoice, silkworm::bytes_of_string(std::string("safeBlockHash")));
```

Comment:
I suggest the following changes to improve readability:
- `silkrpc::db::table` -> `db::table`
- `silkworm::bytes_of_string(std::string("safeBlockHash"))` -> `silkworm::bytes_of_string("safeBlockHash")`
- define constant for "safeBlockHash" at namespace level